### PR TITLE
Addition related to API keys

### PR DESCRIPTION
**Pull request by alallema**
### Breaking Changes

Granular management of API keys is now added to MeiliSearch. New methods have been created to manage this:
- `http.patch` use by `updateKey`
- `client.getKey` get information about a specific API key.
- `client.createKey` create a new API key.
- `client.deleteKey` delete an API key.
- `client.updateKey` update an API key.

### CODE REVIEW

File: src/Endpoints/Delegates/HandlesKeys.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use MeiliSearch\Endpoints\Keys;
+
+/**
+ * @property Keys index
+ */
+trait HandlesKeys
+{
+    public function getKeys(): array
+    {
+        return $this->keys->show();
```
Comment (bidoubiwa) [inline src/Endpoints/Delegates/HandlesKeys.php:16]: where is `show` defined in your code? Should it not be in the Keys endpoint ?
Also for consistency why not `all`?

File: tests/Endpoints/KeysAndPermissionsTest.php [hunk 2]
```diff
@@ -67,8 +50,128 @@ public function testExceptionIfBadKeyProvidedToGetKeys(): void
         $client->getKeys();
     }
 
-    private function getKeys(): array
+    public function testGetKey(): void
```
Comment (bidoubiwa) [inline tests/Endpoints/KeysAndPermissionsTest.php:53]: Missing getKeys test

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

File: tests/Endpoints/KeysAndPermissionsTest.php [hunk 3]
```diff
@@ -67,8 +50,128 @@ public function testExceptionIfBadKeyProvidedToGetKeys(): void
...
+        $this->assertNotNull($response['updatedAt']);
+
+        $this->client->deleteKey($response['key']);
+    }
+
+    public function testDeleteKey(): void
+    {
+        $key = $this->client->createKey(self::INFO_KEY);
+        $this->client->deleteKey($key['key']);
+        $this->expectException(ApiException::class);
```
[the remark this answers is not in the document]
Comment (alallema) [reply tests/Endpoints/KeysAndPermissionsTest.php:168]: It verifies that an error is thrown if you try to get a key that doesn't exist.

File: src/Endpoints/Delegates/HandlesKeys.php (hunk 1 above)
Comment (alallema) [reply src/Endpoints/Delegates/HandlesKeys.php:16]: I hesitated this function is on the global `Endpoint` file in `src/Contract/Endpoint`. I think you are right I going to rewrite it with all

File: tests/Endpoints/KeysAndPermissionsTest.php (hunk 2 above)
Comment (alallema) [reply tests/Endpoints/KeysAndPermissionsTest.php:53]: No it's the first one `testGetKeysDefault`.
This one try to access `key` route without the `masterKey`.

File: tests/Endpoints/KeysAndPermissionsTest.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (bidoubiwa) [reply tests/Endpoints/KeysAndPermissionsTest.php:168]: Oh i see

**Pull request merged**